README.md
# Minder

![Screenshot](data/screenshots/screenshot-light.png "Mind-mapping application for Elementary OS")


## Installation

You will need the following dependencies to build Minder:

* meson
* valac
* debhelper
* libcairo2-dev
* libgranite-dev
* libgtk-3-dev
* libxml2-dev
* libgee-0.8-dev

From the command-line within the top Minder directory, run `./app run` to build and run the application.

To install, run `sudo ./app install` and then run the application from your application launcher.


## Quick Guide to Creating a Mind-Map

When the application is first launched, a main idea node will be displayed in the center of the mind-map canvas. The following actions can be performed on this node:

| Key/Mouse Action | Description |
|------------------|-------------|
| e | Edits the displayed node text (otherwise called the node "name") of the currently selected node. Editable text will be immediately highlighted, so just start typing the new name to change it. To end the text editing, hit either the **Return** key or the **Tab** key. |
| f | Folds the currently selected node, hiding the selected node's tree. |
| z | Zooms out. |
| Z | Zooms in. |
| m | Selects the root node of the current tree. |
| C | Centers the currently selected node in the canvas. |
| Cursor keys | changes the selection to the node in the given direction. |
| Tab | Adds a new child node to the currently selected node. The new node will be immediately editable. |
| Return | Adds a new sibling node of the currently selected node. The new node will be immediately editable. |
| Control-n | Creates a new mind map document. |
| Control-o | Opens an existing document. |
| Control-s | Saves the current state of the document. |
| Control-S | Saves the current mind-map to a specified file. |
| Control-f | Opens the search popover. Entering a string will find all nodes with names that match the strings. Selecting a node in the list will select and bring the node into view on the canvas. |
| Control-z | Undoes the last change. |
| Control-Z | Redoes the last undo change. |
| Right mouse click | Displays a contextual menu listing the available actions that can be taken. |
| Click and drag onto empty space | Moves the selected node to a new position in the selected node's parent. |
| Click and drag onto another node | Detaches the dragged node and attaches it to the node that is dragged over. |

## Property Sidebar

The property sidebar can be shown or hidden by clicking on the gear icon in the upper right corner of the window. This sidebar contains two tabs, one for editing or taking action on the currently selected node. The other tab allows you to modify the layout or theme that is used for the entire mind map.



debian/changelog
com.github.phase1geo.minder (1.0.0) xenial; urgency=low

  * Initial Release.

 -- Trevor Williams <phase1geo@gmail.com>  Sun, 24 Dec 2017 21:05:00 -0500


src/themes/theme_dark.c
/* theme_dark.c generated by valac 0.36.13, the Vala compiler
 * generated from theme_dark.vala, do not modify */

/*
* Copyright (c) 2018 (https://github.com/phase1geo/Minder)
*
* This program is free software; you can redistribute it and/or
* modify it under the terms of the GNU General Public
* License as published by the Free Software Foundation; either
* version 2 of the License, or (at your option) any later version.
*
* This program is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
* General Public License for more details.
*
* You should have received a copy of the GNU General Public
* License along with this program; if not, write to the
* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
* Boston, MA 02110-1301 USA
*
* Authored by: Trevor Williams <phase1geo@gmail.com>
*/

#include <glib.h>
#include <glib-object.h>
#include <stdlib.h>
#include <string.h>
#include <gdk/gdk.h>


#define TYPE_THEME (theme_get_type ())
#define THEME(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), TYPE_THEME, Theme))
#define THEME_CLASS(klass) (G_TYPE_CHECK_CLASS_CAST ((klass), TYPE_THEME, ThemeClass))
#define IS_THEME(obj) (G_TYPE_CHECK_INSTANCE_TYPE ((obj), TYPE_THEME))
#define IS_THEME_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), TYPE_THEME))
#define THEME_GET_CLASS(obj) (G_TYPE_INSTANCE_GET_CLASS ((obj), TYPE_THEME, ThemeClass))

typedef struct _Theme Theme;
typedef struct _ThemeClass ThemeClass;
typedef struct _ThemePrivate ThemePrivate;

#define TYPE_THEME_DARK (theme_dark_get_type ())
#define THEME_DARK(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), TYPE_THEME_DARK, ThemeDark))
#define THEME_DARK_CLASS(klass) (G_TYPE_CHECK_CLASS_CAST ((klass), TYPE_THEME_DARK, ThemeDarkClass))
#define IS_THEME_DARK(obj) (G_TYPE_CHECK_INSTANCE_TYPE ((obj), TYPE_THEME_DARK))
#define IS_THEME_DARK_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), TYPE_THEME_DARK))
#define THEME_DARK_GET_CLASS(obj) (G_TYPE_INSTANCE_GET_CLASS ((obj), TYPE_THEME_DARK, ThemeDarkClass))

typedef struct _ThemeDark ThemeDark;
typedef struct _ThemeDarkClass ThemeDarkClass;
typedef struct _ThemeDarkPrivate ThemeDarkPrivate;

struct _Theme {
	GObject parent_instance;
	ThemePrivate * priv;
};

struct _ThemeClass {
	GObjectClass parent_class;
};

struct _ThemeDark {
	Theme parent_instance;
	ThemeDarkPrivate * priv;
};

struct _ThemeDarkClass {
	ThemeClass parent_class;
};


static gpointer theme_dark_parent_class = NULL;

GType theme_get_type (void) G_GNUC_CONST;
GType theme_dark_get_type (void) G_GNUC_CONST;
enum  {
	THEME_DARK_0_PROPERTY
};
ThemeDark* theme_dark_new (void);
ThemeDark* theme_dark_construct (GType object_type);
Theme* theme_new (void);
Theme* theme_construct (GType object_type);
void theme_set_name (Theme* self, const gchar* value);
void theme_get_color (Theme* self, const gchar* value, GdkRGBA* result);
void theme_set_background (Theme* self, GdkRGBA * value);
void theme_set_foreground (Theme* self, GdkRGBA * value);
void theme_set_root_background (Theme* self, GdkRGBA * value);
void theme_set_root_foreground (Theme* self, GdkRGBA * value);
void theme_set_nodesel_background (Theme* self, GdkRGBA * value);
void theme_set_nodesel_foreground (Theme* self, GdkRGBA * value);
void theme_set_textsel_background (Theme* self, GdkRGBA * value);
void theme_set_textsel_foreground (Theme* self, GdkRGBA * value);
void theme_set_text_cursor (Theme* self, GdkRGBA * value);
void theme_set_link_colors (Theme* self, GdkRGBA* value, int value_length1);
GdkRGBA* theme_get_link_colors (Theme* self, int* result_length1);


ThemeDark* theme_dark_construct (GType object_type) {
	ThemeDark * self = NULL;
	GdkRGBA _tmp0_ = {0};
	GdkRGBA _tmp1_ = {0};
	GdkRGBA _tmp2_ = {0};
	GdkRGBA _tmp3_ = {0};
	GdkRGBA _tmp4_ = {0};
	GdkRGBA _tmp5_ = {0};
	GdkRGBA _tmp6_ = {0};
	GdkRGBA _tmp7_ = {0};
	GdkRGBA _tmp8_ = {0};
	GdkRGBA* _tmp9_;
	GdkRGBA* _tmp10_;
	gint _tmp10__length1;
	GdkRGBA* _tmp11_;
	gint _tmp11__length1;
	GdkRGBA* _tmp12_;
	gint _tmp12__length1;
	GdkRGBA _tmp13_ = {0};
	GdkRGBA _tmp14_;
	GdkRGBA* _tmp15_;
	gint _tmp15__length1;
	GdkRGBA* _tmp16_;
	gint _tmp16__length1;
	GdkRGBA _tmp17_ = {0};
	GdkRGBA _tmp18_;
	GdkRGBA* _tmp19_;
	gint _tmp19__length1;
	GdkRGBA* _tmp20_;
	gint _tmp20__length1;
	GdkRGBA _tmp21_ = {0};
	GdkRGBA _tmp22_;
	GdkRGBA* _tmp23_;
	gint _tmp23__length1;
	GdkRGBA* _tmp24_;
	gint _tmp24__length1;
	GdkRGBA _tmp25_ = {0};
	GdkRGBA _tmp26_;
	GdkRGBA* _tmp27_;
	gint _tmp27__length1;
	GdkRGBA* _tmp28_;
	gint _tmp28__length1;
	GdkRGBA _tmp29_ = {0};
	GdkRGBA _tmp30_;
	GdkRGBA* _tmp31_;
	gint _tmp31__length1;
	GdkRGBA* _tmp32_;
	gint _tmp32__length1;
	GdkRGBA _tmp33_ = {0};
	GdkRGBA _tmp34_;
	GdkRGBA* _tmp35_;
	gint _tmp35__length1;
	GdkRGBA* _tmp36_;
	gint _tmp36__length1;
	GdkRGBA _tmp37_ = {0};
	GdkRGBA _tmp38_;
	GdkRGBA* _tmp39_;
	gint _tmp39__length1;
	GdkRGBA* _tmp40_;
	gint _tmp40__length1;
	GdkRGBA _tmp41_ = {0};
	GdkRGBA _tmp42_;
#line 28 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	self = (ThemeDark*) theme_construct (object_type);
#line 30 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_name ((Theme*) self, "Dark");
#line 33 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#000000", &_tmp0_);
#line 33 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_background ((Theme*) self, &_tmp0_);
#line 34 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "White", &_tmp1_);
#line 34 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_foreground ((Theme*) self, &_tmp1_);
#line 35 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#d4d4d4", &_tmp2_);
#line 35 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_root_background ((Theme*) self, &_tmp2_);
#line 36 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "Black", &_tmp3_);
#line 36 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_root_foreground ((Theme*) self, &_tmp3_);
#line 37 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#64baff", &_tmp4_);
#line 37 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_nodesel_background ((Theme*) self, &_tmp4_);
#line 38 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "Black", &_tmp5_);
#line 38 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_nodesel_foreground ((Theme*) self, &_tmp5_);
#line 39 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#0d52bf", &_tmp6_);
#line 39 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_textsel_background ((Theme*) self, &_tmp6_);
#line 40 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "White", &_tmp7_);
#line 40 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_textsel_foreground ((Theme*) self, &_tmp7_);
#line 41 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "White", &_tmp8_);
#line 41 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_text_cursor ((Theme*) self, &_tmp8_);
#line 44 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp9_ = g_new0 (GdkRGBA, 8);
#line 44 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp10_ = _tmp9_;
#line 44 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp10__length1 = 8;
#line 44 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_set_link_colors ((Theme*) self, _tmp10_, 8);
#line 44 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp10_ = (g_free (_tmp10_), NULL);
#line 45 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp11_ = theme_get_link_colors ((Theme*) self, &_tmp11__length1);
#line 45 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp12_ = _tmp11_;
#line 45 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp12__length1 = _tmp11__length1;
#line 45 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#c6262e", &_tmp13_);
#line 45 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp12_[0] = _tmp13_;
#line 45 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp14_ = _tmp12_[0];
#line 46 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp15_ = theme_get_link_colors ((Theme*) self, &_tmp15__length1);
#line 46 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp16_ = _tmp15_;
#line 46 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp16__length1 = _tmp15__length1;
#line 46 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#f37329", &_tmp17_);
#line 46 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp16_[1] = _tmp17_;
#line 46 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp18_ = _tmp16_[1];
#line 47 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp19_ = theme_get_link_colors ((Theme*) self, &_tmp19__length1);
#line 47 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp20_ = _tmp19_;
#line 47 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp20__length1 = _tmp19__length1;
#line 47 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#f9c440", &_tmp21_);
#line 47 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp20_[2] = _tmp21_;
#line 47 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp22_ = _tmp20_[2];
#line 48 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp23_ = theme_get_link_colors ((Theme*) self, &_tmp23__length1);
#line 48 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp24_ = _tmp23_;
#line 48 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp24__length1 = _tmp23__length1;
#line 48 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#68b723", &_tmp25_);
#line 48 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp24_[3] = _tmp25_;
#line 48 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp26_ = _tmp24_[3];
#line 49 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp27_ = theme_get_link_colors ((Theme*) self, &_tmp27__length1);
#line 49 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp28_ = _tmp27_;
#line 49 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp28__length1 = _tmp27__length1;
#line 49 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#3689e6", &_tmp29_);
#line 49 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp28_[4] = _tmp29_;
#line 49 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp30_ = _tmp28_[4];
#line 50 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp31_ = theme_get_link_colors ((Theme*) self, &_tmp31__length1);
#line 50 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp32_ = _tmp31_;
#line 50 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp32__length1 = _tmp31__length1;
#line 50 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#7a36b1", &_tmp33_);
#line 50 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp32_[5] = _tmp33_;
#line 50 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp34_ = _tmp32_[5];
#line 51 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp35_ = theme_get_link_colors ((Theme*) self, &_tmp35__length1);
#line 51 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp36_ = _tmp35_;
#line 51 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp36__length1 = _tmp35__length1;
#line 51 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#715344", &_tmp37_);
#line 51 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp36_[6] = _tmp37_;
#line 51 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp38_ = _tmp36_[6];
#line 52 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp39_ = theme_get_link_colors ((Theme*) self, &_tmp39__length1);
#line 52 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp40_ = _tmp39_;
#line 52 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp40__length1 = _tmp39__length1;
#line 52 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_get_color ((Theme*) self, "#bbbbbb", &_tmp41_);
#line 52 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp40_[7] = _tmp41_;
#line 52 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	_tmp42_ = _tmp40_[7];
#line 28 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	return self;
#line 310 "theme_dark.c"
}


ThemeDark* theme_dark_new (void) {
#line 28 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	return theme_dark_construct (TYPE_THEME_DARK);
#line 317 "theme_dark.c"
}


static void theme_dark_class_init (ThemeDarkClass * klass) {
#line 25 "/home/trevorw/projects/Minder/src/themes/theme_dark.vala"
	theme_dark_parent_class = g_type_class_peek_parent (klass);
#line 324 "theme_dark.c"
}


static void theme_dark_instance_init (ThemeDark * self) {
}


GType theme_dark_get_type (void) {
	static volatile gsize theme_dark_type_id__volatile = 0;
	if (g_once_init_enter (&theme_dark_type_id__volatile)) {
		static const GTypeInfo g_define_type_info = { sizeof (ThemeDarkClass), (GBaseInitFunc) NULL, (GBaseFinalizeFunc) NULL, (GClassInitFunc) theme_dark_class_init, (GClassFinalizeFunc) NULL, NULL, sizeof (ThemeDark), 0, (GInstanceInitFunc) theme_dark_instance_init, NULL };
		GType theme_dark_type_id;
		theme_dark_type_id = g_type_register_static (TYPE_THEME, "ThemeDark", &g_define_type_info, 0);
		g_once_init_leave (&theme_dark_type_id__volatile, theme_dark_type_id);
	}
	return theme_dark_type_id__volatile;
}





src/minder.c
/* minder.c generated by valac 0.36.12, the Vala compiler
 * generated from minder.vala, do not modify */


#include <glib.h>
#include <glib-object.h>
#include <gtk/gtk.h>
#include <gio/gio.h>
#include <glib/gi18n-lib.h>
#include <stdlib.h>
#include <string.h>


#define TYPE_MINDER (minder_get_type ())
#define MINDER(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), TYPE_MINDER, Minder))
#define MINDER_CLASS(klass) (G_TYPE_CHECK_CLASS_CAST ((klass), TYPE_MINDER, MinderClass))
#define IS_MINDER(obj) (G_TYPE_CHECK_INSTANCE_TYPE ((obj), TYPE_MINDER))
#define IS_MINDER_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), TYPE_MINDER))
#define MINDER_GET_CLASS(obj) (G_TYPE_INSTANCE_GET_CLASS ((obj), TYPE_MINDER, MinderClass))

typedef struct _Minder Minder;
typedef struct _MinderClass MinderClass;
typedef struct _MinderPrivate MinderPrivate;
#define _g_object_unref0(var) ((var == NULL) ? NULL : (var = (g_object_unref (var), NULL)))

struct _Minder {
	GtkApplication parent_instance;
	MinderPrivate * priv;
};

struct _MinderClass {
	GtkApplicationClass parent_class;
};


static gpointer minder_parent_class = NULL;

GType minder_get_type (void) G_GNUC_CONST;
enum  {
	MINDER_0_PROPERTY
};
Minder* minder_new (void);
Minder* minder_construct (GType object_type);
static void minder_real_activate (GApplication* base);
static void _gtk_main_quit_gtk_widget_destroy (GtkWidget* _sender, gpointer self);
gint minder_main (gchar** args, int args_length1);


Minder* minder_construct (GType object_type) {
	Minder * self = NULL;
#line 6 "/home/trevorw/projects/Minder/src/minder.vala"
	self = (Minder*) g_object_new (object_type, "application-id", "com.github.phase1geo.Minder", "flags", G_APPLICATION_FLAGS_NONE, NULL);
#line 5 "/home/trevorw/projects/Minder/src/minder.vala"
	return self;
#line 56 "minder.c"
}


Minder* minder_new (void) {
#line 5 "/home/trevorw/projects/Minder/src/minder.vala"
	return minder_construct (TYPE_MINDER);
#line 63 "minder.c"
}


static void _gtk_main_quit_gtk_widget_destroy (GtkWidget* _sender, gpointer self) {
#line 22 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_main_quit ();
#line 70 "minder.c"
}


static void minder_real_activate (GApplication* base) {
	Minder * self;
	GtkHeaderBar* header = NULL;
	GtkHeaderBar* _tmp0_;
	GtkApplicationWindow* app_window = NULL;
	GtkApplicationWindow* _tmp1_;
	GtkBox* box = NULL;
	GtkBox* _tmp2_;
#line 9 "/home/trevorw/projects/Minder/src/minder.vala"
	self = (Minder*) base;
#line 11 "/home/trevorw/projects/Minder/src/minder.vala"
	_tmp0_ = (GtkHeaderBar*) gtk_header_bar_new ();
#line 11 "/home/trevorw/projects/Minder/src/minder.vala"
	g_object_ref_sink (_tmp0_);
#line 11 "/home/trevorw/projects/Minder/src/minder.vala"
	header = _tmp0_;
#line 12 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_header_bar_set_title (header, _ ("Minder"));
#line 13 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_header_bar_set_subtitle (header, _ ("Mind-Mapping Application"));
#line 14 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_header_bar_set_show_close_button (header, TRUE);
#line 16 "/home/trevorw/projects/Minder/src/minder.vala"
	_tmp1_ = (GtkApplicationWindow*) gtk_application_window_new ((GtkApplication*) self);
#line 16 "/home/trevorw/projects/Minder/src/minder.vala"
	g_object_ref_sink (_tmp1_);
#line 16 "/home/trevorw/projects/Minder/src/minder.vala"
	app_window = _tmp1_;
#line 17 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_window_set_title ((GtkWindow*) app_window, _ ("Minder"));
#line 18 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_window_set_position ((GtkWindow*) app_window, GTK_WIN_POS_CENTER);
#line 19 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_window_set_default_size ((GtkWindow*) app_window, 600, 400);
#line 20 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_window_set_titlebar ((GtkWindow*) app_window, (GtkWidget*) header);
#line 21 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_container_set_border_width ((GtkContainer*) app_window, (guint) 2);
#line 22 "/home/trevorw/projects/Minder/src/minder.vala"
	g_signal_connect ((GtkWidget*) app_window, "destroy", (GCallback) _gtk_main_quit_gtk_widget_destroy, NULL);
#line 29 "/home/trevorw/projects/Minder/src/minder.vala"
	_tmp2_ = (GtkBox*) gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 2);
#line 29 "/home/trevorw/projects/Minder/src/minder.vala"
	g_object_ref_sink (_tmp2_);
#line 29 "/home/trevorw/projects/Minder/src/minder.vala"
	box = _tmp2_;
#line 42 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_container_add ((GtkContainer*) app_window, (GtkWidget*) box);
#line 43 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_widget_show_all ((GtkWidget*) app_window);
#line 45 "/home/trevorw/projects/Minder/src/minder.vala"
	gtk_widget_show ((GtkWidget*) app_window);
#line 9 "/home/trevorw/projects/Minder/src/minder.vala"
	_g_object_unref0 (box);
#line 9 "/home/trevorw/projects/Minder/src/minder.vala"
	_g_object_unref0 (app_window);
#line 9 "/home/trevorw/projects/Minder/src/minder.vala"
	_g_object_unref0 (header);
#line 132 "minder.c"
}


gint minder_main (gchar** args, int args_length1) {
	gint result = 0;
	Minder* app = NULL;
	Minder* _tmp0_;
	gchar** _tmp1_;
	gint _tmp1__length1;
	gint _tmp2_;
#line 51 "/home/trevorw/projects/Minder/src/minder.vala"
	_tmp0_ = minder_new ();
#line 51 "/home/trevorw/projects/Minder/src/minder.vala"
	app = _tmp0_;
#line 53 "/home/trevorw/projects/Minder/src/minder.vala"
	_tmp1_ = args;
#line 53 "/home/trevorw/projects/Minder/src/minder.vala"
	_tmp1__length1 = args_length1;
#line 53 "/home/trevorw/projects/Minder/src/minder.vala"
	_tmp2_ = g_application_run ((GApplication*) app, _tmp1__length1, _tmp1_);
#line 53 "/home/trevorw/projects/Minder/src/minder.vala"
	result = _tmp2_;
#line 53 "/home/trevorw/projects/Minder/src/minder.vala"
	_g_object_unref0 (app);
#line 53 "/home/trevorw/projects/Minder/src/minder.vala"
	return result;
#line 159 "minder.c"
}


int main (int argc, char ** argv) {
#if !GLIB_CHECK_VERSION (2,35,0)
	g_type_init ();
#endif
#line 49 "/home/trevorw/projects/Minder/src/minder.vala"
	return minder_main (argv, argc);
#line 169 "minder.c"
}


static void minder_class_init (MinderClass * klass) {
#line 3 "/home/trevorw/projects/Minder/src/minder.vala"
	minder_parent_class = g_type_class_peek_parent (klass);
#line 3 "/home/trevorw/projects/Minder/src/minder.vala"
	((GApplicationClass *) klass)->activate = (void (*) (GApplication *)) minder_real_activate;
#line 178 "minder.c"
}


static void minder_instance_init (Minder * self) {
}


GType minder_get_type (void) {
	static volatile gsize minder_type_id__volatile = 0;
	if (g_once_init_enter (&minder_type_id__volatile)) {
		static const GTypeInfo g_define_type_info = { sizeof (MinderClass), (GBaseInitFunc) NULL, (GBaseFinalizeFunc) NULL, (GClassInitFunc) minder_class_init, (GClassFinalizeFunc) NULL, NULL, sizeof (Minder), 0, (GInstanceInitFunc) minder_instance_init, NULL };
		GType minder_type_id;
		minder_type_id = g_type_register_static (gtk_application_get_type (), "Minder", &g_define_type_info, 0);
		g_once_init_leave (&minder_type_id__volatile, minder_type_id);
	}
	return minder_type_id__volatile;
}



